tests/test_article_reader.py

```
@@ -270,22 +289,233 @@ def test_run_maps_http_auth_errors_to_browser_boundary():
+        mock.patch("article._validate_remote_url", side_effect=lambda url: url),
+        mock.patch(
+            "article._open_url",
+            side_effect=urllib.error.URLError("connection reset"),
+        ),
     ):
         with pytest.raises(RuntimeError, match="network fetch failed"):
             article.run("https://example.com/down", allow_fetch=True)
 
 
+def _resolver(*addresses: str):
+    def resolve(_host, port, *, type):
```

Ruff (0.16.3)

[error] Function argument `type` is shadowing a Python builtin

(A002)